imap_processing/hi/l1b/hi_l1b.py

```diff
@@ -250,3 +251,47 @@ def compute_coincidence_type_and_time_deltas(dataset: xr.Dataset) -> xr.Dataset:
     out_ds.delta_t_c1c2.values[tof3_valid] = tof_3_ns[tof3_valid]
 
     return out_ds
+
+
+def compute_hae_coordinates(dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Compute HAE latitude and longitude.
+
+    Adds the new variables "hae_latitude" and "hae_longitude", "delta_t_ac1" to
+    the input xarray.Dataset and returns the updated xarray.Dataset.
+
+    Parameters
+    ----------
+    dataset : xarray.Dataset
+        The partial L1B dataset that has had coincidence type, time deltas, and
+        spin phase computed and added to the L1A data.
+
+    Returns
+    -------
+    xarray.Dataset
+        Updated xarray.Dataset with 2 new variables added.
+    """
+    new_data_vars = create_dataset_variables(
+        [
+            "hae_latitude",
+            "hae_longitude",
+        ],
+        len(dataset.epoch),
+        att_manager_lookup_str="hi_de_{0}",
+    )
+    out_ds = dataset.assign(new_data_vars)
```

Comment:
Hmmm, good question. While I do like that the `update` method is an in place operation, it would mean that my `compute_hae_coordinates` function mutates the input which I try to avoid. I could modify the function to just return the dictionary with the new variables and move the code to modify the Dataset into the outer function so that it would look like:
`l1b_dataset.update(compute_hae_coordinates(l1b_dataset))`

I will leave this as is and write a ticket to consider making that change in the general l1b DE processing workflow.